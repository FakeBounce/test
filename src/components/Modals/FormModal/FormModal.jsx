import React, { Component } from 'react';
import { Button } from 'reactstrap';
import { observer } from 'mobx-react';
import { observable } from "mobx";
import { translate } from "common/methods/translations";

import validator from "validator";
import UserModel from "models/UserModel";
import Select from 'elements/Select/Select';
import Input from 'elements/Input/Input';
import Modal from 'elements/Modal/Modal';
import styles from './FormModal.module.scss'
import fieldValidation from 'common/methods/FieldValidation';

@observer
class FormModal extends Component {
  @observable firstName = '';
  @observable lastName = '';
  @observable email = '';
  @observable function_ = '';
  @observable phoneNumber = '';
  @observable profile = '';
  @observable errors = observable({
    firstName: false,
    lastName: false,
    email: false,
    function_: false,
    phoneNumber: false,
    profile: false
  });
  @observable error = false;
  requiredFields = [
    'firstName',
    'lastName',
    'email',
    'function_',
    'phoneNumber',
    'profileId'
  ];

  constructor() {
    super();
    this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
    this.handleLastNameChange = this.handleLastNameChange.bind(this);
    this.handleEmailChange = this.handleEmailChange.bind(this);
    this.handleFunction_Change = this.handleFunction_Change.bind(this);
    this.handlePhoneNumberChange = this.handlePhoneNumberChange.bind(this);
    this.selectedProfile = this.selectedProfile.bind(this);
    this.resetFields = this.resetFields.bind(this);
  }

  resetFields() {
    this.firstName = null;
    this.lastName = null;
    this.email = null;
    this.function_ = null;
    this.phoneNumber = null;
    this.profileId = null;
    this.error = false;
    this.requiredFields.forEach((field) => {
      this.errors[field] = false;
    });
    this.errors.emailFormat = false;
    this.errors.phoneNumberFormat = false;
  }

  saveButtonClick() {
    this.error = false;
    const { handleSaveButtonClick } = this.props;

    this.requiredFields.forEach((field) => {
      if (!this[field]) {
        this.errors[field] = true;
        this.error = true;
      } else {

        switch (field) {
          case 'firstName' : 
            if(!fieldValidation('name', this[field])){
              console.log("FirstName contains bad carac...");
              this.errors[field] = true;
              this.error = true; 
            } else {
              this.errors[field] = false;
            }
            break; 
          case 'lastName' : 
            if(!validator.isAlpha(this[field])) {
              console.log('LastName not good');
              this.errors[field] = true;
              this.error = true; 
            } else {
              this.errors[field] = false;
            }
            break; 
          case 'email' :
              if(!validator.isEmail(this[field])) {
                console.log('Email not good'); 
                this.errors[field] = true;
                this.error = true;
              } else {
                this.errors[field] = false;
              }
              if(!fieldValidation('email', this[field])){
                console.log('Email format not good'); 
                this.errors[field] = true;
                this.errors.emailFormat = true;
                this.error = true;
              } else {
                this.errors[field] = false;
                this.errors.emailFormat = false;
              }
            break;  
          case 'function_' :
            break;  
          case 'phoneNumber' :
              if(!fieldValidation('phoneNumber', this[field])) {
                console.log('phone number with regex  not good'); 
                this.errors[field] = true;
                this.errors.phoneNumberFormat = true;
                this.error = true;
              } else {
                this.errors[field] = false;
                this.errors.phoneNumberFormat = false;
              }
            break;  
          case 'profileId' :
            break;  
          default : 
            break; 

        }
        
      }
    });
    console.log('MyField : ', this.errors);  
    if (!this.error) {
      const newUser = new UserModel(
        null,
        this.firstName,
        this.lastName,
        this.email,
        this.function_,
        this.phoneNumber,
        this.profileId
      );
      handleSaveButtonClick(newUser);
    }
  }

  componentDidMount() {
    this.props.setClick(this.resetFields);
  }

  componentWillUnmount() {

  }

  saveButton() {
    return (
      <Button color="primary" onClick={this.saveButtonClick.bind(this)}>{translate('globals.save')}</Button>
    );
  }

  handleEmailChange(email) {
    this.email = email;
  }

  handleFunction_Change(function_) {
    this.function_ = function_;
  }

  handlePhoneNumberChange(phoneNumber) {
    this.phoneNumber = phoneNumber;
  }

  selectedProfile(profileId) {
    this.profileId = profileId;
  }

  handleFirstNameChange(firstName) {
    this.firstName = firstName;
  }

  handleLastNameChange(lastName) {
    this.lastName = lastName;
  }

  render() {
    const { availableProfiles } = this.props;
    return (
      <Modal
        ModalFooterContent={this.saveButton.bind(this)}
        {...this.props}
        className={`FormModal ${styles.FormModal}`}
      >
        <Input
          inputValue={this.handleFirstNameChange}
          placeholder={translate('myProfile.personalInformation.firstName')}
        />
        <p className={styles.ErrorBox} hidden={!this.errors.firstName}>{translate('globals.required')}</p>
        <Input
          inputValue={this.handleLastNameChange}
          placeholder={translate('myProfile.personalInformation.lastName')}
        />
        <p className={styles.ErrorBox} hidden={!this.errors.lastName}>{translate('globals.required')}</p>
        <Input
          inputValue={this.handleEmailChange}
          placeholder={translate('myProfile.personalInformation.email')}
        />
        <p className={styles.ErrorBox} hidden={!this.errors.email}>{translate('globals.required')}</p>
        <p className={styles.ErrorBox + ' ' + styles.FormatError} hidden={!this.errors.emailFormat}>{translate('myContracts.users.mailError')}</p>
        <Input
          inputValue={this.handleFunction_Change}
          placeholder={translate('myProfile.personalInformation.function')}
        />
        <p className={styles.ErrorBox} hidden={!this.errors.function_}>{translate('globals.required')}</p>
        <Input
          inputValue={this.handlePhoneNumberChange}
          placeholder={translate('myProfile.personalInformation.phoneNumber')}
        />
        <p className={styles.ErrorBox} hidden={!this.errors.phoneNumber}>{translate('globals.required')}</p>
        <p className={styles.ErrorBox + ' ' + styles.FormatError} hidden={!this.errors.phoneNumberFormat}>{translate('myContracts.users.phoneNumberError')}</p>
        <Select
          selectedOption={this.selectedProfile}
          setOptions={availableProfiles}
        />
        <p className={styles.ErrorBox + ' ' + styles.SelectFix} hidden={!this.errors.profileId}>{translate('globals.required')}</p>
      </Modal>
    );
  }
}

export default FormModal;
