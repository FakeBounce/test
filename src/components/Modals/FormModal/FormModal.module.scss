.ErrorBox {
  font-size: 12px;
  color: #FF0000;
  margin-left: 15px;
  position: absolute;
  margin-top: -22px;
}

.FormatError{
  margin-left: 100px;
}

.SelectFix{
  margin-top: -36px;
}

.FormModal {
  input, select {
    width: 98%;
    margin: 5px auto;
    margin-bottom: 22px;
    box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.24) !important;
  }
}