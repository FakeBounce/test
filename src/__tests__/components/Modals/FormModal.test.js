import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import FormModal from "components/Modals/FormModal/FormModal";

configure({ adapter: new Adapter() });

const setClick = jest.fn();
const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  setClick,
  toggle: () => jest.fn()
};

// @TODO Add more tests
describe("Testing FormModal component", () => {
  it("renders as expected without parameters", () => {
    try {
      shallow(<FormModal />);
      // Fail test if above expression doesn't throw anything.
      expect(true).toBe(false);
    } catch (e) {
      expect(e).toBeInstanceOf(TypeError);
      expect(e.message).toBe("this.props.setClick is not a function");
    }
  });

  it("renders as expected with parameters (as closed)", () => {
    const wrapper = shallow(<FormModal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as open)", () => {
    const wrapper = shallow(<FormModal {...defaultTestingProps} isOpen />);
    expect(wrapper).toMatchSnapshot();
  });
});
