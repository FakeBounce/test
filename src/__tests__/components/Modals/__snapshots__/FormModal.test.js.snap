// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormModal component renders as expected with parameters (as closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormModal
    isOpen={false}
    setClick={[Function]}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          inputValue={[Function]}
          placeholder="Prénom"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Nom"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Adresse e-mail"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Le format de l'adresse email n'est pas valide.
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Fonction"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Numéro de téléphone"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Format de numéro erroné
</p>,
        <SelectElement
          selectedOption={[Function]}
          setOptions={undefined}
/>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "FormModal undefined",
      "isOpen": false,
      "setClick": [Function],
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Prénom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Nom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Adresse e-mail",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Le format de l'adresse email n'est pas valide.",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Le format de l'adresse email n'est pas valide.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Fonction",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Numéro de téléphone",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Format de numéro erroné",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Format de numéro erroné",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "selectedOption": [Function],
          "setOptions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            inputValue={[Function]}
            placeholder="Prénom"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Nom"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Adresse e-mail"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Le format de l'adresse email n'est pas valide.
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Fonction"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Numéro de téléphone"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Format de numéro erroné
</p>,
          <SelectElement
            selectedOption={[Function]}
            setOptions={undefined}
/>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "FormModal undefined",
        "isOpen": false,
        "setClick": [Function],
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Prénom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Nom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Adresse e-mail",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Le format de l'adresse email n'est pas valide.",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Le format de l'adresse email n'est pas valide.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Fonction",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Numéro de téléphone",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Format de numéro erroné",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Format de numéro erroné",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selectedOption": [Function],
            "setOptions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing FormModal component renders as expected with parameters (as open) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormModal
    isOpen={true}
    setClick={[Function]}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <InputElement
          inputValue={[Function]}
          placeholder="Prénom"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Nom"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Adresse e-mail"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Le format de l'adresse email n'est pas valide.
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Fonction"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <InputElement
          inputValue={[Function]}
          placeholder="Numéro de téléphone"
/>,
        <p
          className={undefined}
          hidden={true}
>
          Obligatoire
</p>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Format de numéro erroné
</p>,
        <SelectElement
          selectedOption={[Function]}
          setOptions={undefined}
/>,
        <p
          className="undefined undefined"
          hidden={true}
>
          Obligatoire
</p>,
      ],
      "className": "FormModal undefined",
      "isOpen": true,
      "setClick": [Function],
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Prénom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Nom",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Adresse e-mail",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Le format de l'adresse email n'est pas valide.",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Le format de l'adresse email n'est pas valide.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Fonction",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "inputValue": [Function],
          "placeholder": "Numéro de téléphone",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": undefined,
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Format de numéro erroné",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Format de numéro erroné",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "selectedOption": [Function],
          "setOptions": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Obligatoire",
          "className": "undefined undefined",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Obligatoire",
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <InputElement
            inputValue={[Function]}
            placeholder="Prénom"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Nom"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Adresse e-mail"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Le format de l'adresse email n'est pas valide.
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Fonction"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <InputElement
            inputValue={[Function]}
            placeholder="Numéro de téléphone"
/>,
          <p
            className={undefined}
            hidden={true}
>
            Obligatoire
</p>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Format de numéro erroné
</p>,
          <SelectElement
            selectedOption={[Function]}
            setOptions={undefined}
/>,
          <p
            className="undefined undefined"
            hidden={true}
>
            Obligatoire
</p>,
        ],
        "className": "FormModal undefined",
        "isOpen": true,
        "setClick": [Function],
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Prénom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Nom",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Adresse e-mail",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Le format de l'adresse email n'est pas valide.",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Le format de l'adresse email n'est pas valide.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Fonction",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "inputValue": [Function],
            "placeholder": "Numéro de téléphone",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": undefined,
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Format de numéro erroné",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Format de numéro erroné",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selectedOption": [Function],
            "setOptions": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Obligatoire",
            "className": "undefined undefined",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Obligatoire",
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
